### PR TITLE
bpo-45847: Port _gdbm to PY_STDLIB_MOD (GH-29720)

### PR DESCRIPTION
**Pull request by tiran**
Signed-off-by: Christian Heimes <christian@python.org>

https://bugs.python.org/issue45847

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -3366,21 +3374,23 @@ AC_CHECK_HEADERS([db.h], [
...
+
+have_gdbm_dbmliborder=no
+as_save_IFS=$IFS
+IFS=:
+for db in $with_dbmliborder; do
+    AS_CASE([$db],
+      [ndbm], [],
+      [gdbm], [have_gdbm_dbmliborder=yes],
+      [bdb], [],
+      [with_dbmliborder=error]
```
Comment (erlend-aasland) [inline configure.ac:3387]: I'd just put `AC_MSG_ERROR` here, avoiding the `AS_VAR_IF` below, but I'm fine with this variant as well.

File: configure.ac [hunk 2]
```diff
@@ -3366,21 +3374,23 @@ AC_CHECK_HEADERS([db.h], [
...
-      AC_MSG_ERROR([proper usage is --with-dbmliborder=db1:db2:...])
-    fi
-  done
-  IFS=$as_save_IFS
-fi])
+[], [with_dbmliborder=ndbm:gdbm:bdb])
+
+have_gdbm_dbmliborder=no
+as_save_IFS=$IFS
+IFS=:
```
Comment (erlend-aasland) [inline configure.ac:3381]: BTW, why do we save `IFS` here? AFAICS, there's nothing touching it, but I may be mistaken.

**Review by erlend-aasland — APPROVED** [review]

File: configure.ac (hunk 1 above)
Comment (tiran) [reply configure.ac:3387]: I tried it. For unknown reasons it does not work. I get a bunch of shell error messages when the error case is triggered.

File: configure.ac (hunk 2 above)
Comment (tiran) [reply configure.ac:3381]: I'm saving and setting IFS here so the for loop uses ``:`` as a field separator.

File: configure.ac (hunk 1 above)
Comment (erlend-aasland) [reply configure.ac:3387]: `$ man bash  # => /IFS`

Comment (erlend-aasland) [reply configure.ac:3387]:
```
If IFS is unset, the parameters are separated by spaces.  If IFS is null, the parameters are joined without intervening separators.
```

```
The Internal Field Separator that is used for word splitting after expansion and to split lines into words with the read builtin command.  The default value is ``<space><tab><newline>''.
```

File: configure.ac (hunk 2 above)
Comment (erlend-aasland) [reply configure.ac:3381]: Yes, just found it in the man page. Learned something new today.

**Pull request merged**